### PR TITLE
Deprecate `DataFrame.fillna`/`Series.fillna` with `method`

### PR DESCRIPTION
**Pull request by j-bennet**
See https://github.com/dask/dask/issues/10347.

Handles `fillna` deprecations like this one:

```
FAILED dask/dataframe/tests/test_dataframe.py::test_fillna - FutureWarning: DataFrame.fillna with 'method' is deprecated and will raise in a future version. Use obj.ffill() or obj.bfill() instead.
```

- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

The remaining failures are not the same issue.

### CODE REVIEW

File: dask/dataframe/groupby.py [hunk 1]
```diff
@@ -2830,21 +2828,23 @@ def fillna(self, value=None, method=None, limit=None, axis=no_default):
                 "groupby-fillna with value=dict/Series/DataFrame is currently not supported"
             )
         meta = self._meta_nonempty.apply(
-            _fillna_group,
+            _drop_apply,
```
Comment (phofl) [inline dask/dataframe/groupby.py:2831]: This name is a bit confusing to me, can we add fillna somehow?

**Review by phofl — COMMENTED** [review]
small comment, otherwise looks generally good to me.

File: dask/dataframe/groupby.py (hunk 1 above)
Comment (phofl) [reply dask/dataframe/groupby.py:2831]: Ok sounds good

File: dask/dataframe/tests/test_groupby.py [hunk 2]
```diff
@@ -1313,6 +1313,9 @@ def test_fillna(axis, group_keys, method, limit):
     )
     ddf = dd.from_pandas(df, npartitions=2)
 
+    def f(obj, *args, **kwargs):
```
Comment (hendrikmakait) [inline dask/dataframe/tests/test_groupby.py:1316]: I feel like this function makes the test more complicated to read than necessary. It's DRY but not DAMP. 

Can we extract the cases where `f` is used to call `ffill` and `bfill` into dedicated tests `test_{b|f}fill` analogous to `test_fillna`?

**Review by hendrikmakait — COMMENTED** [review]
Overall, the code looks good. I have one open question/suggestion around test readability and one potential new issue we should file.

File: dask/dataframe/core.py [hunk 3]
```diff
@@ -1677,53 +1696,50 @@ def fillna(self, value=None, method=None, limit=None, axis=None):
...
-            }
-            graph = HighLevelGraph.from_collections(name, dsk, dependencies=[self])
-            parts = new_dd_object(graph, name, meta, self.divisions)
-        else:
-            parts = self
-
-        return parts.map_overlap(
-            M.fillna, before, after, method=method, limit=limit, meta=meta
-        )
+            return self.bfill(limit=limit, axis=axis)
```
[the remark this answers is not in the document]
Comment (j-bennet) [reply dask/dataframe/core.py:1712]: This doesn't change the previous behavior:

```python
        if method in ("pad", "ffill"):
            ...
        else:
            method = "bfill"
```

File: dask/dataframe/tests/test_groupby.py (hunk 2 above)
Comment (j-bennet) [reply dask/dataframe/tests/test_groupby.py]: Good point. I updated the tests.

File: dask/dataframe/core.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (hendrikmakait) [reply dask/dataframe/core.py:1712]: This comment is out of scope for this PR, it was more of a general remark about the existing behavior that we may want to fix at some point.

**Review by hendrikmakait — APPROVED** [review]
Thanks, @j-bennet!

**Pull request merged**